app/test/logic/Contacts/VCard/VCard.ts

```diff
@@ -14,41 +15,44 @@ export default {
       throw new Error("No vCard found");
     }
     let vcard = card.containers.vcard[0];
-    if (vcard.entries.fn) {
-      person.name = vcard.entries.fn[0].value;
-    }
+    person.name = vcard.entries.fn?.[0].value || "";
     if (vcard.entries.photo) {
       let photo = vcard.entries.photo[0];
       if (/^b$/i.test(photo.properties.encoding)) { // vCard 3.0
         person.picture = "data:image/" + photo.properties.type?.toLowerCase() + ";base64," + photo.value;
       } else { // vCard 3.0 or 4.0
         person.picture = photo.value;
       }
+    } else {
+      person.picture = null;
     }
-    if (vcard.entries.n) {
-      person.lastName = vcard.entries.n[0].values[0];
-      person.firstName = vcard.entries.n[0].values[1];
-    }
+    person.lastName = vcard.entries.n?.[0].values[0] || "";
+    person.firstName = vcard.entries.n?.[0].values[1] || "";
+    person.emailAddresses.clear();
     if (vcard.entries.email) {
       for (let entry of vcard.entries.email) {
         person.emailAddresses.add(makeContactEntry(entry, "mailto", false));
       }
     }
+    person.phoneNumbers.clear();
```

Comment:
Why the `clear()`s?

Reply:
The intention here is to handle the case where the card was changed on the server, and we need to reflect the changes, and the easiest way to do that is to remove everything we read from the card last time first.